src/TupleTools.jl

```diff
@@ -327,4 +324,13 @@ Inverse permutation of a permutation `p`.
 """
 invperm(p::Tuple{Vararg{Int}}) = sortperm(p)
 
+
+@generated function diff(v::NTuple{N, T}) where {N, T}
+    ex = Expr(:tuple)
+    for k in 1:N-1
+        push!(ex.args, :(v[$k] - v[$(k+1)]))
```

Comment:
also, that should probably be `v[k+1] - v[k]` no?

Reply:
Yes, it is. Thanks for pointing out this mistake.